Assets/Scripts/Guests/Guest.cs

```diff
@@ -224,18 +226,21 @@ private void MoveToNeighbouringTile()
 	/// </summary>
 	private void PathAndMoveTowardsTile(Vector2 targetOffset)
 	{
-        Vector2 myOff = FindCurrentOffset();
-        Debug.Log("find path from "+myOff+" to "+targetOffset);
-        List<Vector2> shortest_path = currentRoom.GetShortestPath(myOff, targetOffset);
-        Debug.Log("path = "+string.Join(",", shortest_path.ToArray()));
+		if(shortestPath == null)
+		{
+			Vector2 myOff = FindCurrentOffset();
+			Debug.Log("find path from "+myOff+" to "+targetOffset);
+			shortestPath = currentRoom.GetShortestPath(myOff, targetOffset);
+			Debug.Log("path = "+string.Join(",", shortestPath.ToArray()));
+		}
 		
-		if(shortest_path.Count < 2)
+		if(shortestPath.Count < 1)

```

Comment:
this is wrong, when Count == 1, removing the only element and then using shortestPath[0] is clearly bad